app/Support/BackupManager.php

```
@@ -0,0 +1,620 @@
+            $target = $this->rootPath . '/' . $relative;
+            $parent = dirname($target);
+            if (!is_dir($parent) && !@mkdir($parent, 0755, true) && !is_dir($parent)) {
+                continue;
+            }
+            $realParent = realpath($parent);
+            if ($realParent === false) {
+                continue;
+            }
+            $realParent = str_replace('\\', '/', $realParent);
+            // The resolved parent must live inside one of the allowed upload roots.
+            $ok = false;
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $ok. Configured minimum length is 3. (undefined)

(ShortVariable)